### ISSUE TITLE
VarCouldBeVal on @Inject lateinit var

### ISSUE THREAD

**Issue by meriouma** [detekt/detekt#4662]
## Expected Behavior

Not getting VarCouldBeVal

## Observed Behavior

Getting VarCouldBeVal

## Context

```kotlin
@MicronautTest
class SomeTest {
    @Inject // <-- VarCouldBeVal targetting this line
    private lateinit var objectMapper: ObjectMapper
}
```

## Your Environment

* Version of detekt used: 1.20.0-RC2
* Version of Gradle used (if applicable): 7.4.1

**Comment by cortinico** [comment]
Is this a regression from 1.19.0?

**Comment by meriouma** [comment]
Seems to be. Just tested with 1.19.0 and the error is not reported.

**Comment by meriouma** [comment]
I believe this is related to this PR adding more validations : https://github.com/detekt/detekt/pull/4424

**Comment by BraisGabin** [comment]
This is working as expected. You should use `ignoreAnnotated: ['Inject']` to ignore those.
